rmw_connextdds_common/src/common/rmw_context.cpp

```diff
@@ -435,6 +442,21 @@ rmw_context_impl_t::finalize()
 {
   rmw_ret_t rc_exit = RMW_RET_OK;
 
+  if (this->discovery_options && this->base) {
```

Comment:
I fixed up this function according to your comments @asorbini in 5ae99e9.

However, I also removed the check about `this->base`, because it seemed to me that if `this->discovery_options` was not nullptr, then we should finalize it, regardless of base. @mxgrey can you comment on whether or not that's correct?

Reply:
I noticed that and it was the right call imho.

Reply:
Yeah, the check was there because if `base` was `nullptr` then there was no way to deallocate, but since the upstream API was changed to put the allocator inside of `rmw_discovery_options` itself, we don't depend on still having `base` anymore. LGTM.